Content.Server/Body/Systems/InternalsSystem.cs

```diff
@@ -224,59 +227,35 @@ private short GetSeverity(InternalsComponent component)
         return 1;
     }
 
-    public Entity<GasTankComponent>? FindBestGasTank(EntityUid internalsOwner, InternalsComponent component)
+    public Entity<GasTankComponent>? FindBestGasTank(Entity<HandsComponent?, InventoryComponent?, ContainerManagerComponent?> user)
     {
         // Prioritise
         // 1. back equipped tanks
         // 2. exo-slot tanks
         // 3. in-hand tanks
         // 4. pocket/belt tanks
-        InventoryComponent? inventory = null;
-        ContainerManagerComponent? containerManager = null;
 
-        if (_inventory.TryGetSlotEntity(internalsOwner, "back", out var backEntity, inventory, containerManager) &&
+        if (!Resolve(user.Owner, ref user.Comp1, ref user.Comp2, ref user.Comp3))
+            return null;
+
+        if (_inventory.TryGetSlotEntity(user.Owner, "back", out var backEntity, user.Comp2, user.Comp3) &&
             TryComp<GasTankComponent>(backEntity, out var backGasTank) &&
             _gasTank.CanConnectToInternals(backGasTank))
         {
             return (backEntity.Value, backGasTank);
         }
 
-        if (_inventory.TryGetSlotEntity(internalsOwner, "suitstorage", out var entity, inventory, containerManager) &&
+        if (_inventory.TryGetSlotEntity(user.Owner, "suitstorage", out var entity, user.Comp2, user.Comp3) &&
             TryComp<GasTankComponent>(entity, out var gasTank) &&
             _gasTank.CanConnectToInternals(gasTank))
         {
             return (entity.Value, gasTank);
         }
 
-        var tanks = new List<Entity<GasTankComponent>>();
-
-        foreach (var hand in _hands.EnumerateHands(internalsOwner))
-        {
-            if (TryComp(hand.HeldEntity, out gasTank) && _gasTank.CanConnectToInternals(gasTank))
-                tanks.Add((hand.HeldEntity.Value, gasTank));
-        }
-
-        if (tanks.Count > 0)
+        foreach (var item in _inventory.GetHandOrInventoryEntities((user.Owner, user.Comp1, user.Comp2)))
         {
-            tanks.Sort((x, y) => y.Comp.Air.TotalMoles.CompareTo(x.Comp.Air.TotalMoles));
-            return tanks[0];
-        }
-
-        if (Resolve(internalsOwner, ref inventory, false))
-        {
-            var enumerator = new InventorySystem.ContainerSlotEnumerator(internalsOwner, inventory.TemplateId, _protoManager, _inventory, SlotFlags.POCKET | SlotFlags.BELT);
-
-            while (enumerator.MoveNext(out var container))
-            {
-                if (TryComp(container.ContainedEntity, out gasTank) && _gasTank.CanConnectToInternals(gasTank))
-                    tanks.Add((container.ContainedEntity.Value, gasTank));
-            }
-
-            if (tanks.Count > 0)
-            {
-                tanks.Sort((x, y) => y.Comp.Air.TotalMoles.CompareTo(x.Comp.Air.TotalMoles));
-                return tanks[0];
```

Comment:
This no longer sorts by moles, but IMO its better to have a consistent order anyways, because otherwise toggling the internals off and on again could unexpectedly switch tanks.